kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -158,20 +158,17 @@ public static Collection<String> getSearchNodesToQuery(
     // step 1 - find all snapshots that match time window and partition
     ScopedSpan snapshotsToSearchSpan =
         Tracing.currentTracer().startScopedSpan("KaldbDistributedQueryService.snapshotsToSearch");
-    Set<String> snapshotsToSearch =
-        snapshotMetadataStore
-            .getCached()
-            .stream()
-            .filter(
-                snapshotMetadata ->
-                    containsDataInTimeRange(
-                            snapshotMetadata.startTimeEpochMs,
-                            snapshotMetadata.endTimeEpochMs,
-                            queryStartTimeEpochMs,
-                            queryEndTimeEpochMs)
-                        && isSnapshotInPartition(snapshotMetadata, partitions))
-            .map(snapshotMetadata -> snapshotMetadata.name)
-            .collect(Collectors.toSet());
+    Set<String> snapshotsToSearch = new HashSet<>(256);
```

Comment:
We should probably leave this default until we have a better idea of the performance implications. With the default load factor this still isn't high enough to prevent a resize for recent queries.

If we want to keep a magic number here, it should be documented on how we arrived at the value so people adjusting it in the future understand the motivation.

Reply:
Switched to default value. I think tweaking this number to be a small value, would help remove some allocations.